### ISSUE TITLE
Following azure-simple getting started documentation produces an error

### ISSUE THREAD

**Issue by nmiodice** [microsoft/cobalt#146]
### Description
The following set of commands:
```bash
export TF_VAR_remote_state_account=...
export TF_VAR_remote_state_container=...
export ARM_ACCESS_KEY=...

$ terraform init -backend-config "storage_account_name=${TF_VAR_remote_state_account}" -backend-config "container_name=${TF_VAR_remote_state_container}"
$ terraform plan
```

Produces the following error:
```bash
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.azurerm_client_config.current: Refreshing state...
data.azurerm_client_config.current: Refreshing state...

------------------------------------------------------------------------

Error: Error running plan: 3 error(s) occurred:

* module.keyvault.output.keyvault_id: Resource 'azurerm_key_vault.keyvault' not found for variable 'azurerm_key_vault.keyvault.id'
* module.keyvault.output.keyvault_uri: Resource 'azurerm_key_vault.keyvault' not found for variable 'azurerm_key_vault.keyvault.vault_uri'
* module.keyvault.output.keyvault_name: Resource 'azurerm_key_vault.keyvault' not found for variable 'azurerm_key_vault.keyvault.name'
```

### Acceptance Criteria
Reference: [Done-Done Checklist] (https://github.com/Microsoft/code-with-engineering-playbook/blob/master/Engineering/BestPractices/DoneDone.md)
- [ ] Should ‹testable condition that should be satisfied›
- [ ] Should ‹testable condition that should be satisfied›
- [ ] …

Also, here are a few points that need to be addressed:

1. Constraint 1;
1. Constraint 2;
1. Constraint 3.

### Resources
*Technical Design Document*
*Mockups*

### Tasks
*Stories are intended to be completed in a single sprint; if task breakdown creates addition work then team should discuss promoting the Story to an Epic.
Reference: [Minimal Valuable Slices] (https://github.com/Microsoft/code-with-engineering-playbook/blob/master/Engineering/BestPractices/MinimalSlices.md)*

Reference: [How to Write Better Tasks] (http://agilebutpragmatic.blogspot.com/2012/04/splitting-story-into-tasks-how-to-write.html)

**_Assignee should break down work into tasks here_**

**Comment by TechnicallyWilliams** [comment]
Thanks for making this official! 

**Git Bash vs WSL issue?**

Within Git Bash, I ran into the same error. However, within WSL I was able to overcome the error with a fresh install of Terraform and Go.  Still can't say for sure that the issue is shell based. We'll have to do a bit more testing before updating the README and closing this out .

For both shells, I was logged in as a service-principal within the AZ CLI using Terraform v.0.11.13 and Go v.1.11.

**Comment by nmiodice** [comment]
@TechnicallyWilliams I am on WSL and have the same issues that you see in `gitbash`.

I'm wondering if your fresh install is contributing to some environment-level changes. I'll deep dive on this today.

Thanks for the added context

**Comment by nmiodice** [comment]
@TechnicallyWilliams I have made a breakthrough in debugging...

I had been running the following:
`. ../../../.env`

Which would essentially execute a series of:
```bash
FOO=BAR
```

I found, though, that this method of setting variables does not make the variables available to sub-processes (https://stackoverflow.com/questions/1158091/defining-a-variable-with-or-without-export) so the variables were not available to any terraform child processes.

The solution can be scripted easily. I'll find a 1-liner that we can add to the `README.md` file.
